pinot-core/src/main/java/org/apache/pinot/core/plan/CombinePlanNode.java

```diff
@@ -73,10 +76,16 @@ public CombinePlanNode(List<PlanNode> planNodes, QueryContext queryContext, Exec
     _streamObserver = streamObserver;
   }
 
-  @SuppressWarnings({"rawtypes", "unchecked"})
   @Override
   public BaseCombineOperator run() {
+    try (InvocationScope scope = Tracing.getTracer().createScope(CombinePlanNode.class)) {
+      return getCombineOperator(scope);
+    }
+  }
+
+  private BaseCombineOperator getCombineOperator(InvocationScope scope) {
```

Comment:
It’s interesting - my intention in creating separate interfaces for lifecycle management and data capture was so the interface for data capture could be passed around without worrying whether the scope got closed somewhere (without downcasting that is, someone would have to go out of their way to close a scope safely if they receive InvocationRecording) - perhaps the intent was not clear. Not that it’s particularly important.

Reply:
I see. Modified to not pass the scope around